src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -625,6 +626,9 @@ R"(#include <stdio.h>
                 // ASR assigns return variable only once at the end of function
                 alloc_return_var = true;
             }
+            if (ASRUtils::is_character(*m_target_type)) {
+                alloc_target = indent + "if (!" + target + ") " + target + " = (char *) malloc(40 * sizeof(char));\n";
```

Comment:
Why is this 40? That seems it will break.

Reply:
We are currently fixing it to 40 (everywhere), but we need to define some strong support later that handles reallocation like lists.

Reply:
I think its just some pre-allocation we are doing at declaration. Yes, we need to have some flag which will help us figure out if a particular `char*` is allocated or not. If its not at a particular point of usage then allocate it and then use it.